automation/taskcluster/lib/tasks.py

```diff
@@ -49,18 +64,24 @@ def craft_build_task(self, module_name, gradle_tasks, subtitle='', run_coverage=
 
         command = ' && '.join(cmd for cmd in (gradle_command, post_gradle_command) if cmd)
 
+        features = {}
+        if component is not None:
+            features['chainOfTrust'] = True
+        elif any(scope.startswith('secrets:') for scope in scopes):
+            features['taskclusterProxy'] = True
+
         return self._craft_build_ish_task(
             name='Android Components - Module {} {}'.format(module_name, subtitle),
             description='Execute Gradle tasks for module {}'.format(module_name),
             command=command,
             scopes=scopes,
-            artifacts=artifacts
+            artifacts=taskcluster_artifacts
         )
 
-    def craft_wait_on_builds_task(self, dependencies):
+    def craft_barrier_task(self, dependencies):
```

Comment:
I renamed this to a more generic name as I didn't want to duplicate it. Snapshot beetmover needs to wait on all builds tasks, while github release's beetmover need to wait for all signing tasks. I believe barrier is a better name regardless as we might need to add more tasks in the future at some point.

Reply:
Why do we want to wait for all signing tasks before we do pushes?
I'm :+1: on waiting for builds, since it's possible for a subset of modules to fail to compile.
However, for beetmoving, can't we push as soon as signing for a module finishes? Is there a possible scenario for a subset of signing jobs to fail?

Reply:
This is a very good point. I've amended the signing tasks to depend on the barrier instead. Let's parallelize signing and beetmover after barrier upon builds is completed 👍